app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -287,25 +287,45 @@ MinionPoller = {
   },
 
   handleAdminUpdate: function(admin) {
```

Comment:
Instead of handling everything here, we could extract all the migration type into something like `handleAdminMigrationUpdate` and `handleAdminMaintenanceUpdate`. Trying to stay sane... :sweat_smile:

Reply:
yeah, good idea! this is still very raw ;) I'll try to get the back-end things working and then make the UI prettier